pkg/services/ngalert/store/deltas_test.go

```diff
@@ -240,14 +240,14 @@ func TestCalculateChanges(t *testing.T) {
 		require.Len(t, changes.AffectedGroups[sourceGroupKey], len(inDatabase))
 	})
 
-	t.Run("should fail when submitted rule has UID that does not exist in db", func(t *testing.T) {
```

Comment:
Let's add some tests here covering the new expected behaviour of `CalculateChanges`